Failure in SearchableSnapshotDirectoryTests.testClearCache

**Build scan**: https://gradle-enterprise.elastic.co/s/ortf7bsvdpbzq

**Repro line**: `./gradlew ':x-pack:plugin:searchable-snapshots:test' --tests "org.elasticsearch.index.store.SearchableSnapshotDirectoryTests.testClearCache" -Dtests.seed=720D9DF9013086A4 -Dtests.security.manager=true -Dtests.locale=es-CR -Dtests.timezone=Asia/Ulan_Bator -Druntime.java=16 -Dtests.fips.enabled=true`

**Reproduces locally?**: No

**Applicable branches**: master, 7.x, 7.12

**Failure history**: Seems to have started on 2/24/21. See [build stats](https://build-stats.elastic.co/app/kibana#/discover?_g=(refreshInterval:(pause:!t,value:0),time:(from:now-6M,mode:quick,to:now))&_a=(columns:!(project),filters:!(('$state':(store:appState),meta:(alias:!n,disabled:!f,index:e58bf320-7efd-11e8-bf69-63c8ef516157,key:project,negate:!f,params:(query:elasticsearch,type:phrase),type:phrase,value:elasticsearch),query:(match:(project:(query:elasticsearch,type:phrase)))),('$state':(store:appState),meta:(alias:!n,disabled:!f,index:e58bf320-7efd-11e8-bf69-63c8ef516157,key:test,negate:!f,params:(query:testClearCache,type:phrase),type:phrase,value:testClearCache),query:(match:(test:(query:testClearCache,type:phrase)))),('$state':(store:appState),meta:(alias:!n,disabled:!f,index:e58bf320-7efd-11e8-bf69-63c8ef516157,key:message,negate:!f,params:(query:'Number%20of%20files%20(0)%20mismatch',type:phrase),type:phrase,value:'Number%20of%20files%20(0)%20mismatch'),query:(match:(message:(query:'Number%20of%20files%20(0)%20mismatch',type:phrase))))),index:e58bf320-7efd-11e8-bf69-63c8ef516157,interval:auto,query:(language:lucene,query:''),sort:!(time,desc))).

**Failure excerpt**:
```

java.lang.AssertionError: Number of files (0) mismatch, got : [] |  
-- | --
  | Expected: (a value greater than <0> and a value less than or equal to <5>) |  
  | but: a value greater than <0> <0> was equal to <0>


at __randomizedtesting.SeedInfo.seed([720D9DF9013086A4:BCD2A73583A3C9E4]:0) |  
-- | --
  |   | at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18) |  
  |   | at org.junit.Assert.assertThat(Assert.java:956) |  
  |   | at org.elasticsearch.index.store.SearchableSnapshotDirectoryTests.assertListOfFiles(SearchableSnapshotDirectoryTests.java:904) |  
  |   | at org.elasticsearch.index.store.SearchableSnapshotDirectoryTests.testClearCache(SearchableSnapshotDirectoryTests.java:740)


```

--- a/x-pack/plugin/searchable-snapshots/src/test/java/org/elasticsearch/index/store/SearchableSnapshotDirectoryTests.java
+++ b/x-pack/plugin/searchable-snapshots/src/test/java/org/elasticsearch/index/store/SearchableSnapshotDirectoryTests.java
@@ -650,7 +650,6 @@ public class SearchableSnapshotDirectoryTests extends AbstractSearchableSnapshot
         });
     }
 
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/69914")
     public void testClearCache() throws Exception {
         try (CacheService cacheService = defaultCacheService()) {
             cacheService.start();
@@ -728,7 +727,8 @@ public class SearchableSnapshotDirectoryTests extends AbstractSearchableSnapshot
 
                     try (IndexInput input = directory.openInput(fileInfo.physicalName(), randomIOContext())) {
                         assertThat(input.length(), equalTo((long) fileLength));
-                        final int start = between(0, fileLength - 1);
+                        // we can't just read footer as that is not served from cache
+                        final int start = between(0, fileLength - CodecUtil.footerLength() - 1);
                         final int end = between(start + 1, fileLength);
 
                         input.seek(start);

